profiles/seccomp/seccomp.go

```
@@ -6,28 +6,64 @@ import (
 	"encoding/json"
 	"fmt"
 
+	"github.com/docker/docker/pkg/stringutils"
 	"github.com/docker/engine-api/types"
+	"github.com/docker/engine-api/types/versions/v1p24"
 	"github.com/opencontainers/specs/specs-go"
+	libseccomp "github.com/seccomp/libseccomp-golang"
 )
 
 //go:generate go run -tags 'seccomp' generate.go
 
 // GetDefaultProfile returns the default seccomp profile.
 func GetDefaultProfile(rs *specs.Spec) (*specs.Seccomp, error) {
-	return setupSeccomp(DefaultProfile(rs))
+	return setupSeccomp(DefaultProfile(), rs)
 }
 
 // LoadProfile takes a file path and decodes the seccomp profile.
-func LoadProfile(body string) (*specs.Seccomp, error) {
-	var config types.Seccomp
+func LoadProfile(body string, rs *specs.Spec) (*specs.Seccomp, error) {
+	var (
+		config types.Seccomp
+		err    error
+	)
 	if err := json.Unmarshal([]byte(body), &config); err != nil {
 		return nil, fmt.Errorf("Decoding seccomp profile failed: %v", err)
 	}
+	// try with the old profile
+	// TODO(runcom): this is not going to work properly:
+	// if I use Version == 0 in a new format seccomp profile, what's going to
+	// happen?
+	if config.Version == 0 {
```

I think I like supporting `name` and `names` better, as then all v0 configs are valid v1 which makes it easier for people to upgrade, and silently ignoring is evil...